js/src/components/free-listings/configure-product-listings/checkErrors.js

```diff
@@ -27,15 +27,41 @@ const checkErrors = (
 
 	if (
 		values.shipping_rate === 'flat' &&
-		( shippingRates.length < finalCountryCodes.length ||
-			shippingRates.some( ( el ) => el.rate < 0 ) )
+		( values.shipping_country_rates.length < finalCountryCodes.length ||
+			values.shipping_country_rates.some( ( el ) => el.rate < 0 ) )
 	) {
 		errors.shipping_rate = __(
 			'Please specify shipping rates for all the countries. And the estimated shipping rate cannot be less than 0.',
 			'google-listings-and-ads'
 		);
 	}
 
+	/**
+	 * Check offer free shipping.
+	 */
+	if (
+		values.offer_free_shipping === undefined &&
+		values.shipping_country_rates.some( isNonFreeFlatShippingRate )
+	) {
+		errors.offer_free_shipping = __(
+			'Please select an option for offering free shipping.',
+			'google-listings-and-ads'
+		);
+	}
+
+	if (
+		values.offer_free_shipping === true &&
+		values.shipping_country_rates.every(
+			( shippingRate ) =>
+				shippingRate.options.free_shipping_threshold === undefined
+		)
+	) {
```

Comment:
🚧 

It's unable to save when selecting `automatic` or `manual` shipping. And it would be nicer to have test cases to cover this.

![Kapture 2022-03-11 at 14 08 56](https://user-images.githubusercontent.com/17420811/157812325-1312c825-76f7-4efc-a8c7-5239ebe38f60.gif)

Reply:
🤦‍♂️ Forgot about the settings up there. Thanks @eason9487. I have fixed it in [`d290236` (#1309)](https://github.com/woocommerce/google-listings-and-ads/pull/1309/commits/d290236cc85afdbec3e9d69da5aaabe648b92579), with tests in it.